x-pack/plugins/enterprise_search/public/applications/app_search/components/meta_engine_creation/meta_engine_creation.tsx

```diff
@@ -0,0 +1,175 @@
+    MetaEngineCreationLogic
+  );
+
+  useEffect(() => {
+    fetchIndexedEngineNames();
+  }, []);
+
+  return (
+    <div data-test-subj="MetaEngineCreation">
+      <SetPageChrome trail={[META_ENGINE_CREATION_TITLE]} />
+      <EuiPageHeader>
+        <EuiPageHeaderSection>
+          <EuiTitle size="l">
+            <h1>{META_ENGINE_CREATION_TITLE}</h1>
+          </EuiTitle>
+          <EuiText>{META_ENGINE_CREATION_FORM_META_ENGINE_DESCRIPTION}</EuiText>
+          <EuiText>{META_ENGINE_CREATION_FORM_DOCUMENTATION_DESCRIPTION}</EuiText>
+        </EuiPageHeaderSection>
+      </EuiPageHeader>
+      <EuiPageBody>
+        <FlashMessages />
+        <EuiPanel>
+          <EuiForm>
+            <form
+              data-test-subj="MetaEngineCreationForm"
+              onSubmit={(e) => {
+                e.preventDefault();
+                submitEngine();
+              }}
+            >
+              <EuiTitle>
+                <EuiText>{META_ENGINE_CREATION_FORM_TITLE}</EuiText>
+              </EuiTitle>
+              <EuiSpacer />
+              <EuiFlexGroup>
+                <EuiFlexItem>
+                  <EuiFormRow
+                    data-test-subj="MetaEngineCreationNameFormRow"
+                    label={META_ENGINE_CREATION_FORM_ENGINE_NAME_LABEL}
+                    helpText={
+                      name.length > 0 && rawName !== name ? (
+                        <>
+                          {SANITIZED_NAME_NOTE} <strong>{name}</strong>
+                        </>
+                      ) : (
+                        ALLOWED_CHARS_NOTE
+                      )
+                    }
+                    fullWidth
+                  >
+                    <EuiFieldText
+                      name="engine-name"
+                      value={rawName}
+                      onChange={(event) => setRawName(event.currentTarget.value)}
+                      autoComplete="off"
+                      fullWidth
+                      data-test-subj="MetaEngineCreationNameInput"
+                      placeholder={META_ENGINE_CREATION_FORM_ENGINE_NAME_PLACEHOLDER}
+                      autoFocus
+                    />
+                  </EuiFormRow>
+                </EuiFlexItem>
+              </EuiFlexGroup>
+              <EuiSpacer />
+              <EuiFormRow label={META_ENGINE_CREATION_FORM_ENGINE_SOURCE_ENGINES_LABEL} fullWidth>
+                <EuiComboBox
+                  data-test-subj="MetaEngineCreationSourceEnginesInput"
+                  options={indexedEngineNames.map(engineNameToComboBoxOption)}
+                  selectedOptions={selectedIndexedEngineNames.map(engineNameToComboBoxOption)}
+                  onChange={(options) => {
+                    setSelectedIndexedEngineNames(options.map(comboBoxOptionToEngineName));
+                  }}
+                />
+              </EuiFormRow>
+              <EuiSpacer />
+              {selectedIndexedEngineNames.length > maxEnginesPerMetaEngine && (
```

Comment:
Looks like this branching is uncovered - can we add a unit test for this callout?

Reply:
That should also cover the uncovered line 60 in `constants.tsx` with the `META_ENGINE_CREATION_FORM_MAX_SOURCE_ENGINES_WARNING_TITLE` fn

Reply:
Dumb question: Are you just eyeballing these uncovered branches or is there a tool you're using to work out specifically what's covered/not covered.

Reply:
From the Enterprise Search plugin root (`x-pack/plugins/enteprise_search/`) you can run `sh jest.sh {YOUR_COMPONENT_DIR}`, or in this case:
```sh
# Note: tab to complete should work to autocomplete folders
sh jest.sh sh jest.sh public/applications/app_search/components/meta_engine_creation/
```

That should give you a code coverage report that looks like this:

<img width="1147" alt="" src="https://user-images.githubusercontent.com/549407/109538210-ad773d80-7a74-11eb-8ba3-ac92533b064b.png">

Docs: https://github.com/elastic/kibana/tree/master/x-pack/plugins/enterprise_search#unit-tests - the Getting Started guide also has the above command listed

I also know Jason uses a `--watch` flag to re-run tests whenever changes to a file is made - I don't personally (I prefer to make changes all at once and run tests when I'm ready) but you can definitely append `--watch` to the `sh jest.sh` command to get dynamic code coverage updates.

Reply:
Thanks!